test/extensions/filters/http/jwt_authn/authenticator_test.cc

```diff
@@ -403,17 +410,113 @@ TEST_F(AuthenticatorTest, TestNoForwardPayloadHeader) {
 
   MockUpstream mock_pubkey(mock_factory_ctx_.cluster_manager_, PublicKey);
   auto headers = Http::TestHeaderMapImpl{{"Authorization", "Bearer " + std::string(GoodToken)}};
-  MockAuthenticatorCallbacks mock_cb;
-  EXPECT_CALL(mock_cb, onComplete(_)).WillOnce(Invoke([](const Status& status) {
+  std::function<void(const Status&)> mock_cb = [](const Status& status) {
     ASSERT_EQ(status, Status::Ok);
-  }));
-  auth_->verify(headers, &mock_cb);
+  };
+  auto tokens = filter_config_->getExtractor().extract(headers);
+  auth_->verify(headers, std::move(tokens), std::move(mock_cb));
 
   // Test when forward_payload_header is not set, the output should NOT
   // contain the sec-istio-auth-userinfo header.
   EXPECT_FALSE(headers.has("sec-istio-auth-userinfo"));
 }
 
+// This test verifies that path match prefix, and JWT authenticates.
+TEST_F(AuthenticatorTest, TestPrefixPathMatch) {
+  auto& rule = *proto_config_.add_rules();
+  rule.mutable_match()->set_prefix("/abc");
```

Comment:
I thought in this test "rule" is not used.  we create auth() directly without using any rule.

Reply:
see next comment.